lib/elixir/src/elixir_with.erl

```diff
@@ -57,3 +78,21 @@ build_cases([], DoExpr) ->
   DoExpr;
 build_cases([{_, Meta, _} = Expr | Rest], DoExpr) ->
   {'__block__', Meta, [Expr, build_cases(Rest, DoExpr)]}.
+
+build_else(Meta, WithCases, ElseClauses) ->
+  Result = {'result', Meta, nil},
+  Clauses = [
+    {'->', Meta, [[{{'^', Meta, [?ref(Meta)]}, Result}], Result]}
+    | ElseClauses
+  ] ++ [build_raise(Meta)],
+  {'case', Meta, [WithCases, [{do, Clauses}]]}.
+
+reference(Meta) ->
```

Comment:
```
with {:ok, width} <- Map.fetch(opts, :width),
     {:ok, height} <- %WithResult{...} do 
  {:ok, width * height}
else
  :error -> :error
end
```

Than, it is possibly to fake in theory, than wrapping both to `{:ok, ..}` and `{:error, ..}` is cleaner, because it is not possible to fake from inside of with.